### PR TITLE
Custom field layout

### PR DESCRIPTION
**Pull request by jasone**

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -202,16 +215,48 @@ public static final class AutoConstants {
...
+    }
+    private static final boolean kUseCustomAprilTagFieldLayout = false;
+    public static final AprilTagFieldLayout kAprilTagFieldLayout = kUseCustomAprilTagFieldLayout
+      ? loadCustomAprilTagFieldLayout()
+      : AprilTagFields.k2024Crescendo.loadAprilTagLayoutField();
+
+    public static final double kMaxX = kAprilTagFieldLayout.getFieldLength();
+
+    private static final Pose3d kBlueSpeakerAprilTagPose =
+    kAprilTagFieldLayout.getTags().get(7).pose;
```
Comment (SamueljWilson) [inline src/main/java/frc/robot/Constants.java:252]: Should be 6 not 7

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -202,16 +215,48 @@ public static final class AutoConstants {
...
+    public static final Translation2d kBlueSpeaker =
+      new Translation2d(kBlueSpeakerAprilTagPose.getX(), kBlueSpeakerAprilTagPose.getY());
+
+    private static final Pose3d kRedSpeakerAprilTagPose =
+      kAprilTagFieldLayout.getTags().get(4).pose;
```
Comment (SamueljWilson) [inline src/main/java/frc/robot/Constants.java]: Should be 3 not 4

File: src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (SamueljWilson) [inline src/main/java/frc/robot/Constants.java]: Should be 6 not 7

**Review by SamueljWilson — CHANGES_REQUESTED** [review]

**Review by SamueljWilson — APPROVED** [review]

File: src/main/java/frc/robot/Constants.java (hunk 2 above)
Comment (jasone) [reply src/main/java/frc/robot/Constants.java]: Addressed in the context of #59.

File: src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (jasone) [reply src/main/java/frc/robot/Constants.java]: Addressed in the context of #59.

**Review by SamueljWilson — APPROVED** [review]

**Pull request merged**